### PR TITLE
idl reduction/splitting of TryConstruct and XCDR

### PR DESCRIPTION
**Pull request by ClaytonCalabrese**
IDL was generating files too large causing memory errors on build machines. This PR should fix that issue by splitting the idl into 3 different files.  Each of these files is less than half the size of the previous combined generated TSImpl file.

### CODE REVIEW

**Comment by mitza-oci** [comment]
If there is duplication across IDL files, why not use `#include`?

**Comment by ClaytonCalabrese** [comment]
> If there is duplication across IDL files, why not use `#include`?

I'm not sure how exactly the memory management works with these idl files, and so without a way to directly test it, I was trying to play it safe by avoiding #includes.

Do #includes not stay in memory when building idl?

**Comment by mitza-oci** [comment]
> 
> 
> > If there is duplication across IDL files, why not use `#include`?
> 
> I'm not sure how exactly the memory management works with these idl files, and so without a way to directly test it, I was trying to play it safe by avoiding #includes.
> 
> Do #includes not stay in memory when building idl?

The goal is to reduce the size of each individual TypeSupportImpl.cpp.  Not to reduce the size of the preprocessed IDL.

File: tests/DCPS/Compiler/xcdr/xcdr2.idl [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  @id(3) NestedUnion union_nested;
+  @id(4) StringSeq sequence_field2;
+};
+
+@mutable
+struct ModifiedMixedMutableStruct {
+  @id(3) NestedUnion union_nested;
+  @id(4) StringSeq sequence_field2;
+  @id(1) MutableStructWithExplicitIDs struct_nested;
+};
```
Comment (iguessthislldo) [inline tests/DCPS/Compiler/xcdr/xcdr2.idl:134]: Missing EOF newline

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

File: tests/DCPS/Compiler/TryConstruct/TryConstructBaseTypes.idl [hunk 2]
```diff
@@ -0,0 +1,155 @@
+
+#include <tao/ShortSeq.pidl>
+#include <tao/UShortSeq.pidl>
+#include <tao/StringSeq.pidl>
+#include <tao/WStringSeq.pidl>
+
+//enum for use in struct, union, and sequence
+//due to issue 1900, enums in C++11 need an extra value beyond each case when used as a union discriminator
```
Comment (sonndinh) [inline tests/DCPS/Compiler/TryConstruct/TryConstructBaseTypes.idl:8]:
```suggestion
// enum for use in struct, union, and sequence
// due to issue 1900, enums in C++11 need an extra value beyond each case when used as a union discriminator
```
Similar comment for the remaining lines of this file (and other files).

File: tests/DCPS/Compiler/TryConstruct/TryConstructBaseTypes.idl [hunk 3]
```diff
@@ -0,0 +1,155 @@
+
```
Comment (sonndinh) [inline tests/DCPS/Compiler/TryConstruct/TryConstructBaseTypes.idl:1]:
```suggestion
```

File: tests/DCPS/Compiler/TryConstruct/TryConstruct2.idl [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+  @extensibility(MUTABLE)
+  union DefaultDiscrimUnion switch (@try_construct(USE_DEFAULT) EnumType) {
+    case VALUE1: short s1;
+    case VALUE2: short s2;
+  };
+
+    @mutable
+  struct DiscardStructString1 {
+    str64 str64_d;
+  };
```
Comment (sonndinh) [inline tests/DCPS/Compiler/TryConstruct/TryConstruct2.idl:45]:
```suggestion
  @mutable
  struct DiscardStructString1 {
    str64 str64_d;
  };
```

File: tests/DCPS/Compiler/TryConstruct/TryConstruct3.idl [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+  @mutable
+  struct NestedStructTest1{
+    NestedStruct2 ns;
+  };
+
+  @mutable
+  struct NestedStructTest2{
+    @try_construct(USE_DEFAULT) NestedStruct3 ns;
+  };
```
Comment (sonndinh) [inline tests/DCPS/Compiler/TryConstruct/TryConstruct3.idl:79]:
```suggestion
  @mutable
  struct NestedStructTest1 {
    NestedStruct2 ns;
  };

  @mutable
  struct NestedStructTest2 {
    @try_construct(USE_DEFAULT) NestedStruct3 ns;
  };
```

**Review by sonndinh — COMMENTED** [review]
Can the IDL files be split and named so that it's easier to tell what's inside each resulting file?

**Review by iguessthislldo — CHANGES_REQUESTED** [review]
Could you rename `mutabletypes.idl` to `mutable_types.idl`?

File: tests/DCPS/Compiler/TryConstruct/AnonTypes.idl [hunk 6]
```diff
@@ -0,0 +1,104 @@
+#include <TryConstructBaseTypes.idl>
+
+module TryCon {
+
+  @topic
+  @mutable
+  struct AnonSeqStruct {
+    sequence <EnumType, 3> AnonEnumSeqBound;
```
Comment (sonndinh) [inline tests/DCPS/Compiler/TryConstruct/AnonTypes.idl:8]:
```suggestion
    sequence<EnumType, 3> AnonEnumSeqBound;
```
For other sequences too.

File: tests/DCPS/Compiler/TryConstruct/AnonTypes.idl [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    @try_construct(TRIM) sequence <ShortSeqUnbound> AnonShortSeqUnboundUnbound;
+    @try_construct(TRIM) sequence <ShortSeqUnbound, 2> AnonSeqShortSeqUnboundBound;
+    @try_construct(TRIM) sequence <@try_construct(TRIM) ShortSeqBound> AnonSeqShortSeqBoundUnbound;
+    @try_construct(TRIM) sequence <@try_construct(TRIM) ShortSeqBound, 2> AnonSeqShortSeqBoundBound;
+  };
+
+  @topic
+  @mutable
+  struct AnonSeqStructDefault {
+    @try_construct(USE_DEFAULT) sequence < EnumType, 2> AnonEnumSeqBound;
```
Comment (sonndinh) [inline tests/DCPS/Compiler/TryConstruct/AnonTypes.idl:48]:
```suggestion
    @try_construct(USE_DEFAULT) sequence <EnumType, 2> AnonEnumSeqBound;
```

File: tests/DCPS/Compiler/TryConstruct/AnonTypes.idl [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+  @topic
+  @mutable
+  struct AnonArrStructTrim {
+    // string
+    str20 AnonStringArr @try_construct(TRIM) [3];
+    // wstring
+    wstr20 AnonWideStringArr @try_construct(TRIM) [3];
+    // array
+    str20ArrayTrim AnonArrayArr [3];
```
Comment (sonndinh) [inline tests/DCPS/Compiler/TryConstruct/AnonTypes.idl:86]:
```suggestion
    str20ArrayTrim AnonArrayArr[3];
```

**Review by sonndinh — APPROVED** [review]

**Review by iguessthislldo — APPROVED** [review]

**Pull request merged**